### ISSUE TITLE
Investigate odd behavior with type evaluation

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3395]
This is a tracking bug. For original, see https://github.com/microsoft/pylance-release/issues/2745.

```python
import random


class apple:
    rot_way: str

    def __init__(self) -> None:
        self.rot_time = "3 days"

    def rot(self, *args):
        print(f"will rot in {self.rot_time}")
        return random.choice(["ok", "eaten by bugs"])

    def get_rot_way(self):
        return self.rot_way


class grow:
    location: str
    spped: str


def func():
    fruit = apple()

    grow_recipe: grow = None

    if grow_recipe is None:
        if fruit.rot() == "ok":
            grow_recipe = grow_recipe()
            grow_recipe.location = "Lin Garden"
            grow_recipe.spped = "3 days"

    if grow_recipe is not None:
        print(f"apple will reborn at {grow_recipe.location}")
        return

    idx = [i for i in [] if i == "dummy"]

    fruit.get_rot_way()
```

If the line starting with `idx = ...` is commented out, the expression `fruit.get_rot_way` evaluates to a callable as expected, but otherwise evaluates to `Never`.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.242, which I just published. It will also be included in the next release of pylance.
